back/app/controllers/omniauth_callback_controller.rb

```diff
@@ -19,6 +19,25 @@ def create
     end
   end
 
+  def failure
```

Comment:
`failure` and `logout` look like added just because I've moved `auth_callback` under the private section (as it's not an action).

Reply:
Thanks for doing this :+1: